src/ngraph/runtime/gpu/pass/gpu_layout.cpp

```diff
@@ -79,6 +81,114 @@ namespace ngraph
                         reshape->set_op_annotations(op_annotations);
                     }
                 }
+
+                template <>
+                void GPULayout::LAYOUT_DECL(ngraph::op::TopK)
+                {
+                    auto topk = static_cast<ngraph::op::TopK*>(node.get());
+                    auto topk_axis = topk->get_top_k_axis();
+                    auto topk_k = topk->get_k();
+                    auto parent_node = topk->get_argument(0);
+
+                    auto in_shape = topk->get_input_shape(0);
+                    auto out_shape = in_shape;
+                    out_shape[topk_axis] = topk_k;
+                    size_t ndim = static_cast<size_t>(in_shape.size());
+
+                    AxisVector reshape_axis_order = get_ordered_axis_vector(ndim);
+                    reshape_axis_order.erase(reshape_axis_order.begin() + topk_axis);
+                    reshape_axis_order.push_back(topk_axis);
+
+                    Shape pre_reshape_out;
+                    for (size_t j = 0; j < ndim; j++)
+                    {
+                        pre_reshape_out.push_back(in_shape[reshape_axis_order[j]]);
+                    }
+
+                    Shape pre_2d_reshape_out(2);
+                    pre_2d_reshape_out[1] = pre_reshape_out[ndim - 1];
+                    pre_2d_reshape_out[0] = static_cast<size_t>(
+                        ngraph::shape_size(pre_reshape_out) / pre_2d_reshape_out[1]);
+
+                    auto pre_reshape = make_shared<ngraph::op::Reshape>(
+                        parent_node, reshape_axis_order, pre_reshape_out);
+
+                    AxisVector axis_order = get_ordered_axis_vector(ndim);
+                    auto pre_2d_reshape = make_shared<ngraph::op::Reshape>(
+                        pre_reshape, axis_order, pre_2d_reshape_out);
+
+                    insert_new_node_between(parent_node, node, pre_reshape);
+                    parent_node = topk->get_argument(0);
+                    insert_new_node_between(parent_node, node, pre_2d_reshape);
```

Comment:
remove line 121 and do `insert_new_node_between(pre_reshape, node, pre_2d_reshape);` in the same way you construct the pre_2d_reshape above